src/ai/backend/gateway/groupconfig.py

```diff
@@ -0,0 +1,252 @@
+import logging
+import re
+from typing import Any, Tuple
+
+import sqlalchemy as sa
+from aiohttp import web
+import aiohttp_cors
+import trafaret as t
+
+from ai.backend.common import validators as tx
+from ai.backend.common import msgpack
+from ai.backend.common.logging import BraceStyleAdapter
+
+from .auth import auth_required
+from .exceptions import (
+    InvalidAPIParameters, DotfileCreationFailed,
+    DotfileNotFound, DotfileAlreadyExists,
+    GenericForbidden, GroupNotFound
+)
+from .manager import READ_ALLOWED, server_status_required
+from .types import CORSOptions, Iterable, WebMiddleware
+from .utils import check_api_params
+
+from ..manager.models import (
+    groups,
+    query_group_dotfiles,
+    verify_dotfile_name,
+    MAXIMUM_DOTFILE_SIZE,
+)
+
+log = BraceStyleAdapter(logging.getLogger('ai.backend.gateway.dotfile'))
+
+
+@server_status_required(READ_ALLOWED)
+@auth_required
+@check_api_params(t.Dict(
+    {
+        tx.AliasedKey(['group', 'groupId', 'group_id']): tx.UUID | t.String,
+        t.Key('domain', default=None): t.String | t.Null,
+        t.Key('data'): t.String(max_length=MAXIMUM_DOTFILE_SIZE),
+        t.Key('path'): t.String,
+        t.Key('permission'): t.Regexp(r'^[0-7]{3}$', re.ASCII),
+    }
+))
+async def create(request: web.Request, params: Any) -> web.Response:
+    log.info('CREATE DOTFILE (group: {0})', params['group'])
+    dbpool = request.app['dbpool']
+    if not request['is_superadmin'] and not request['is_admin']:
+        raise GenericForbidden('Only superadmin or admin can create dotfiles')
```

Comment:
This can be achived by `@admin_required` decorator. Also, group dotfiles should not be created by the admins of a domain where the group does not belong to.